index.js

```diff
@@ -11,50 +11,101 @@ var couch2pg = require('./libs/couch2pg/importer')(
       db, couchdb,
       env.couch2pgDocLimit,
       env.couch2pgChangesLimit),
-    xmlforms = require('./libs/xmlforms/extractor')(db);
-
-var batchLoop = function() {
-  var completeBatch = function(lastSeq) {
-    return couch2pg.storeSeq(lastSeq)
-      .then(function() {
-        log.info('Batch completed up to ' + lastSeq);
-      });
-  };
-
-  log.info('Beginning couch2pg and xmlforms batch');
-
-  return couch2pg.importBatch()
-  .then(function(changes) {
-    var allDocs = changes.deleted.concat(changes.edited);
-    log.debug('There are ' + allDocs.length + ' changes');
-
-    if (allDocs.length > 0) {
-      return xmlforms.extract(allDocs)
-      .then(function() {
-        return completeBatch(changes.lastSeq);
-      })
-      .then(batchLoop);
+    xmlforms = require('./libs/xmlforms/updater')(db);
+
+var fallback = 0;
+var sleepMs = function(errored) {
+  if (errored) {
+    var fallbackMs = fallback * 1000 * 60;
+    if (fallbackMs < env.sleepMs) {
+      fallback++;
+      return fallbackMs;
+    } else {
+      return env.sleepMs;
+    }
+  } else {
+    fallback = 0;
+    return env.sleepMs;
+  }
+};
+
+var migrateCouch2pg = function() {
+  return couch2pgMigrator(env.postgresqlUrl)();
+};
+var migrateXmlforms = function() {
+  return xmlformsMigrator(env.postgresqlUrl)();
+};
+
+var delayLoop = function(loopFn, errored) {
+  return new Promise(function(resolve) {
+    var ms = sleepMs(errored);
+    log.info('Run '+(errored ? 'errored' : 'complete') + '. Next run at ' + new Date(new Date().getTime() + ms));
+    if (ms === 0) {
+      resolve();
     } else {
-      return completeBatch(changes.lastSeq);
+      setInterval(resolve, ms);
     }
   });
 };
 
 var run = function() {
-  log.info('Beginning couch2pg and xmlforms run');
+  log.info('Beginning couch2pg and xmlforms run at ' + new Date());
```

Comment:
Yes. Loglevel (the library we're using) does this with a plugin, I just haven't got around to doing that yet. I should raise a ticket.